googleapiclient/discovery.py

```diff
@@ -1110,19 +1125,36 @@ def methodResource(self):
                                method.__get__(self, self.__class__))
 
   def _add_next_methods(self, resourceDesc, schema):
-    # Add _next() methods
-    # Look for response bodies in schema that contain nextPageToken, and methods
-    # that take a pageToken parameter.
+    # Add _next() methods iff one of the names 'pageToken' or 'nextPageToken'
+    # occurs among the fields of both the method's response type and either the
+    # method's request (query parameters) or request body.
     if 'methods' in resourceDesc:
       for methodName, methodDesc in six.iteritems(resourceDesc['methods']):
-        if 'response' in methodDesc:
-          responseSchema = methodDesc['response']
-          if '$ref' in responseSchema:
-            responseSchema = schema.get(responseSchema['$ref'])
-          hasNextPageToken = 'nextPageToken' in responseSchema.get('properties',
-                                                                   {})
-          hasPageToken = 'pageToken' in methodDesc.get('parameters', {})
-          if hasNextPageToken and hasPageToken:
-            fixedMethodName, method = createNextMethod(methodName + '_next')
+        nextPageToken = pageTokenName(
+            methodProperties(methodDesc, schema, 'response'))
+        if nextPageToken:
+          isPageTokenParameter = True
+          pageToken = pageTokenName(methodDesc.get('parameters', {}))
+          if not pageToken:
+            isPageTokenParameter = False
+            pageToken = pageTokenName(
+                methodProperties(methodDesc, schema, 'request'))
+          if pageToken:
+            fixedMethodName, method = createNextMethod(
+                methodName + '_next', pageToken, nextPageToken,
+                isPageTokenParameter)
             self._set_dynamic_attr(fixedMethodName,
                                    method.__get__(self, self.__class__))
+
+
+pageTokenNames = ('pageToken', 'nextPageToken')
```

Comment:
Module-level constants go at the top and should be `UPPER_SNAKE_CASE`. Also, there's no need for this to be public so please do `_PAGE_TOKEN_FIELD_NAMES`.